src/_pytest/_code/source.py

```diff
@@ -136,6 +136,23 @@ def deindent(self) -> "Source":
         newsource.lines[:] = deindent(self.lines)
         return newsource
 
+    def isparseable(self, deindent: bool = True) -> bool:
+        """ return True if source is parseable, heuristically
+            deindenting it by default.
+        """
+        if deindent:
+            source = str(self.deindent())
+        else:
+            source = str(self)
+        try:
+            ast.parse(source)
+        except KeyboardInterrupt:
+            raise
+        except Exception:
+            return False
```

Comment:
`KeyboardInterrupt` is a `BaseException` but not an `Exception`, so we don't need this clause.

Reply:
According to the docs for `exec`, all that can happen here is `(SyntaxError, ValueError, TypeError)` - so why not catch that instead?

Reply:
Done, thanks for the suggestions. Applied the same change to #6408